src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -91,8 +91,7 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             }
 
             var version = ResolveVersion(detectionResult.PlatformVersion);
-            detectionResult.PlatformVersion = version;
-            return detectionResult;
+            return new PlatformDetectorResult(detectionResult.Platform, version);
```

Comment:
I was hoping for this work item to get triaged before we start to implement. For my perspective having this immutable is not of a big concern. The more richer we make these detector results, this constructor pattern will not be friendly.